runtime/platform.c

```diff
@@ -144,12 +145,26 @@ uintnat caml_mem_round_up_pages(uintnat size)
 #define MAP_FAILED 0
 #endif
 
+#ifdef DEBUG
+static struct lf_skiplist mmap_blocks = {NULL};
+static caml_plat_mutex mmap_blocks_lock = CAML_PLAT_MUTEX_INITIALIZER;
+#endif
+
 void* caml_mem_map(uintnat size, uintnat alignment, int reserve_only)
 {
   uintnat alloc_sz = caml_mem_round_up_pages(size + alignment);
   void* mem;
   uintnat base, aligned_start, aligned_end;
 
+#ifdef DEBUG
+  if (mmap_blocks.head == NULL) {
+    caml_plat_lock(&mmap_blocks_lock);
+    if (mmap_blocks.head == NULL)
+      caml_lf_skiplist_init(&mmap_blocks);
+    caml_plat_unlock(&mmap_blocks_lock);
+  }
+#endif
```

Comment:
Hah - I had considered this, but worried about its being brittle 🙂 Maybe on this occasion I was being overly cautious - it's in the debug runtime, and the lock is essentially pointless at the moment, so perhaps with more comments it would be fine to note that it isn't racy.

Reply:
I thought about brittleness. Let's simply `CAMLassert(caml_domain_alone())`, this should be enough.

Reply:
Thanks, this looks fine to me for debug code (I'd have `<= 1` instead of `== 0` but whatever).

Reply:
I changed that when squashing!